pysat/tests/instrument_test_class.py

```diff
@@ -0,0 +1,253 @@
+                            inst.name, inst.tag, inst.sat_id)))
+            raise merr
+
+    @pytest.mark.second
+    @pytest.mark.download
+    @pytest.mark.parametrize("clean_level", ['none', 'dirty', 'dusty',
+                                             'clean'])
+    def test_load(self, clean_level, inst):
+        """Check that instruments load at each cleaning level."""
+        # make sure download was successful
+        if len(inst.files.files) > 0:
+            try:
+                inst.clean_level = clean_level
+                target = 'Fake Data to be cleared'
+                inst.data = [target]
+                start = inst._test_dates[inst.sat_id][inst.tag]
+                inst.load(date=start)
+                # Make sure fake data is cleared
+                assert target not in inst.data
+                # If cleaning not used, something should be in the file
+                # Not used for clean levels since cleaning may remove all data
+                if clean_level == "none":
+                    assert not inst.empty
+                # For last parametrized clean_level, remove files
+                if clean_level == "clean":
+                    remove_files(inst)
+            except Exception as merr:
+                # Let users know which instrument is failing, as instrument
+                # list is opaque
+                print(' '.join(('\nProblem with loading:', inst.platform,
+                                inst.name, inst.tag, inst.sat_id)))
+                raise merr
+        else:
+            pytest.skip("Download data not available")
+
+    @pytest.mark.download
+    def test_remote_file_list(self, inst):
+        """Check if optional list_remote_files routine exists and is callable.
+        """
+        try:
+            name = '_'.join((inst.platform, inst.name))
+            if hasattr(getattr(pysat.instruments, name), 'list_remote_files'):
+                assert callable(inst.remote_file_list)
+            else:
+                pytest.skip("remote_file_list not available")
+        except Exception as merr:
+            print(' '.join(('\nProblem with checking:', inst.platform,
+                            inst.name, inst.tag, inst.sat_id)))
+            raise merr
+
+    @pytest.mark.no_download
+    def test_download_warning(self, inst):
+        """Check that instruments without download support have a warning."""
+        start = inst._test_dates[inst.sat_id][inst.tag]
+        try:
+            with warnings.catch_warnings(record=True) as war:
+                inst.download(start, start)
+            assert len(war) >= 1
+            categories = [war[j].category for j in range(0, len(war))]
+            assert UserWarning in categories
+        except Exception as merr:
+            # Let users know which instrument is failing, as instrument
+            # list is opaque
+            print(' '.join(('\nProblem with checking:', inst.platform,
```

Comment:
Can this print statement be added to the raise message instead?

Reply:
That would be a lot cleaner in tracker down errors.  Thanks!